redfish-core/lib/pcie_slots.hpp

```diff
@@ -139,13 +139,74 @@ inline void
             "xyz.openbmc_project.Inventory.Item.PCIeDevice"});
 }
 
+inline void
+    linkAssociatedProcessor(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                            const std::string& pcieSlotPath, size_t index)
+{
+    crow::connections::systemBus->async_method_call(
+        [asyncResp, pcieSlotPath,
+         index](const boost::system::error_code ec,
+                const std::variant<std::vector<std::string>>& endpoints) {
+            if (ec)
+            {
+                if (ec.value() == EBADR)
+                {
+                    // This PCIeSlot have no processor association.
+                    BMCWEB_LOG_DEBUG << "No processor association found";
+                    return;
+                }
+                BMCWEB_LOG_ERROR << "DBUS response error" << ec.message();
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            const std::vector<std::string>* value =
+                std::get_if<std::vector<std::string>>(&(endpoints));
+
+            if (value == nullptr)
+            {
+                BMCWEB_LOG_DEBUG << "Error getting processor association!";
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            if ((*value).size() < 1)
+            {
+                BMCWEB_LOG_DEBUG << "No association found for processor";
```

Comment:
why does a nullptr generate an internalError but this doesn't?

Reply:
nullptr denotes that we did not get the type that we expect as endpoint hence error but having processor associated with a slot is not always true so in case there is none just a log and return without populating that link.

This was the thought process I had in mind. Please correct me if I got that wrong.

Reply:
I was thinking if there is no processor, the association query would fail.  If the association query does not fail, yet you get here and there are no processors - doesn't that indicate some kind of logic bug?

Reply:
Agree, fixed.